lib/org_date_rangeable.rb

```diff
@@ -0,0 +1,27 @@
+module OrgDateRangeable
+  def monthly_range(org:, start_date: Date.today.end_of_month, end_date: Date.today.end_of_month)
+    raise ArgumentError.new('missing org') unless org.present?
+    raise ArgumentError.new('missing start_date') unless start_date.respond_to?(:end_of_month)
```

Comment:
Instead of raising an exception if the parameter doesn't respond to an expected method, you can force the type by explicitly calling the `to_[type]` method. 

By _soft testing_ the type of an object, the code should be more flexible and robust.

``` ruby 
# bad
def yell_name(name)
  raise ArgumentError, "Not a string" unless name.respond_to?(:uppercase)
  puts name.uppercase
end

# good
def yell_name(name)
  puts name.to_s.uppercase
end
```

Reply:
... in this case, just use `start_date.to_date`

Reply:
thanks for sharing these notes. Since the query will run with any argument passed (except for org that send ':id'), I think is best removing duck typing checks. What do you think?